sigmastate/src/main/scala/sigmastate/interpreter/ProverInterpreter.scala

```diff
@@ -346,36 +319,50 @@ trait ProverInterpreter extends Interpreter with AttributionCore {
     t.withPolynomial(q).copy(children=newChildren)
     */
 
-
-
     case su: UnprovenSchnorr =>
-      if (su.simulated) {
-        // Step 5 (simulated leaf -- complete the simulation)
-        assert(su.challengeOpt.isDefined)
-        val (fm, sm) = DLogInteractiveProver.simulate(su.proposition, su.challengeOpt.get)
-        UncheckedSchnorr(su.proposition, Some(fm), su.challengeOpt.get, sm)
-      } else {
-        // Step 6 (real leaf -- compute the commitment a)
-        val (r, commitment) = DLogInteractiveProver.firstMessage(su.proposition)
-        su.copy(commitmentOpt = Some(commitment), randomnessOpt = Some(r))
+      //Steps 5 & 6: pull out commitment from the hints bag, otherwise, compute the commitment(if the node is real),
+      // or simulate it (if the node is simulated)
```

Comment:
Comment better written as 
`Steps 5 & 6: first try pulling out commitment from the hints bag. If it exists proceed with it, otherwise, compute the commitment (if the node is real) or simulate it (if the node is simulated).`

Reply:
done